Issue #24686 Added 'verbose' argument to find_hung_process.py

Elley-Folks opened the pull request:
Refs #24686 this adds the '--verbose' or '-v' argument for find_hung_process.py that additionally outputs the commands used in the script.

Elley-Folks committed:
Added 'verbose' argument to find_hung_process.py

permcody reviewed (changes requested):
Looks very good! I do have two requests though:

1. Typically we want to print the relevant output before we run the task. If the tasks take a long time, it can be really odd to see "Running command ...." after that command has already run. Just move your lines up before the corresponding execution lines and we should be good to go!
2. Testing... We should always have a test for everything we do. We recently added some tests for this script here: [Tests](https://github.com/idaholab/moose/blob/next/python/find_hung_process/tests). Go ahead and add another one (can be RunApp type test). You'll want to test for the right output pattern.

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/1589049/) on 0bf63d1 wanted to post the following:


## Framework coverage

<table>
  <thead>
    <tr>
      <th></th>
      <th>288f85</th>
      <th colspan=4>#24687 0bf63d</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>85.40%</td>
      <td>85.40%</td>
      <td>-0.00%</td>
      <td>-</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>88934</td>
      <td>88933</td>
      <td>-1</td>
      <td>0</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>15199</td>
      <td>15200</td>
      <td>+1</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/24687/coverage_diff/framework)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/24687/coverage/framework)


## Modules coverage

Inconsistent report tags were found between the head and base reports.
This can happen when reports are missing from either the head or the base.

Inconsistent tags:
  module-contact-parallel
  module-heat_conduction-parallel

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/24687/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/24687/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/24687/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/24687/coverage/contact)
- [`electromagnetics`](https://mooseframework.inl.gov/docs/PRs/24687/coverage/electromagnetics)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/24687/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/24687/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/24687/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/24687/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/24687/coverage/geochemistry)
- [`heat_conduction`](https://mooseframework.inl.gov/docs/PRs/24687/coverage/heat_conduction)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/24687/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/24687/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/24687/coverage/navier_stokes)
- [`optimization`](https://mooseframework.inl.gov/docs/PRs/24687/coverage/optimization)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/24687/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/24687/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/24687/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/24687/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/24687/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/24687/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/24687/coverage/richards)
- [`scalar_transport`](https://mooseframework.inl.gov/docs/PRs/24687/coverage/scalar_transport)
- [`solid_properties`](https://mooseframework.inl.gov/docs/PRs/24687/coverage/solid_properties)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/24687/coverage/stochastic_tools)
- [`tensor_mechanics`](https://mooseframework.inl.gov/docs/PRs/24687/coverage/tensor_mechanics)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/24687/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/24687/coverage/xfem)

</details>



This comment will be updated on new commits.

Elley-Folks committed:
Added test and moved up output lines.

Added verbose_test to check if --verbose or -v is being output correctly by script.

Moved up output lines find_hung_process.py as requested.

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/1590370/) on ab624c6 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/24687/site)

This comment will be updated on new commits.

permcody reviewed (changes requested):
- Factor out the duplicate string for the `qstat -n` command into a new string variable
- Move your `qstat` printing up a few more lines before the run command
- Rerun your bad_mpi program to get more interesting stack traces to check in.
- Change the double quotes to single quotes in your test specification

Elley-Folks committed:
Updated find_hung_process.py, cache and test

Refactored the qstat command in generate_traces, eliminated duplicate command. Got a more interesting cache file, and updated test to run that cache file.

Closes Issue #24686

Elley-Folks committed:
Updated tests so command runs -v

Updated command in [verbose_test] to correctly run cache file and actually use the -v command.

Closes Issue #24686
